onnxruntime/core/graph/graph_utils.h

```diff
@@ -187,5 +187,73 @@ void FinalizeNodeFusion(Graph& graph, Node& first_node, Node& second_node);
 */
 void FinalizeNodeFusion(Graph& graph, const std::vector<std::reference_wrapper<Node>>& nodes, Node& replacement_node);
 
+
+/*
+Find the first input edge of a node that the source node's operator type, version, and domain match the given values.
+@returns nullptr when not found.
+*/
+const Node::EdgeEnd*
+FindFirstInputEdge(
+    const Node& node,
+    const std::string& op_type,
+    const std::initializer_list<ONNX_NAMESPACE::OperatorSetVersion>& versions,
+    const std::string& domain);
+
+/*
+Find the input edge of a node for a specified input index.
+@returns nullptr when not found.
+*/
+const Node::EdgeEnd* GetInputEdge(const Node& node, int arg_index);
+
+/*
+Find the source node of an input edge for a specified input index.
+@returns nullptr when not found.
+*/
+const Node* GetInputNode(const Node& node, int arg_index);
+
+
+/*
+Expected edge information for matching.
```

Comment:
Let's add some more information about this struct, what is its purpose etc.
It seems that it is tailored for either input or output edges -- by looking at it, it is not immediately clear which of the two, but I see that dst_arg_index and no source index, so it cannot be generic as is.
If this is indeed the case, we should rename this to reflect whether it focuses on input or output edges.

Reply:
I'll update the struct to support both input and output edge.